apps/opencs/model/world/infoselectwrapper.cpp

```diff
@@ -0,0 +1,893 @@
+        case '8': mFunctionName = Function_NotFaction; break;
+        case '9': mFunctionName = Function_NotClass; break;
+        case 'A': mFunctionName = Function_NotRace; break;
+        case 'B': mFunctionName = Function_NotCell; break;
+        case 'C': mFunctionName = Function_NotLocal; break;
+        default:  mFunctionName = Function_None; break;
+    }
+}
+
+void CSMWorld::ConstInfoSelectWrapper::readRelationType()
+{
+    char relationIndex = mConstSelect.mSelectRule[RelationIndexOffset];
+
+    switch (relationIndex)
+    {
+        case '0': mRelationType = Relation_Equal; break;
+        case '1': mRelationType = Relation_NotEqual; break;
+        case '2': mRelationType = Relation_Greater; break;
+        case '3': mRelationType = Relation_GreaterOrEqual; break;
+        case '4': mRelationType = Relation_Less; break;
+        case '5': mRelationType = Relation_LessOrEqual; break;
+        default:  mRelationType = Relation_None;
+    }
+}
+
+void CSMWorld::ConstInfoSelectWrapper::readVariableName()
+{
+    if (mConstSelect.mSelectRule.size() >= VarNameOffset)
+        mVariableName = mConstSelect.mSelectRule.substr(VarNameOffset);
+    else
+        mVariableName.clear();
+}
+
+void CSMWorld::ConstInfoSelectWrapper::updateHasVariable()
+{
+    switch (mFunctionName)
+    {
+        case Function_Global:
+        case Function_Local:
+        case Function_Journal:
+        case Function_Item:
+        case Function_Dead:
+        case Function_NotId:
+        case Function_NotFaction:
+        case Function_NotClass:
+        case Function_NotRace:
+        case Function_NotCell:
+        case Function_NotLocal:
+            mHasVariable = true;
+            break;
+
+        default:
+            mHasVariable = false;
+            break;
+    }
+}
+
+void CSMWorld::ConstInfoSelectWrapper::updateComparisonType()
+{
+    switch (mFunctionName)
+    {
+        // Boolean
+        case Function_NotId:
+        case Function_NotFaction:
+        case Function_NotClass:
+        case Function_NotRace:
+        case Function_NotCell:
+        case Function_NotLocal:
+        case Function_PcExpelled:
+        case Function_PcCommonDisease:
+        case Function_PcBlightDisease:
+        case Function_SameSex:
+        case Function_SameRace:
+        case Function_SameFaction:
+        case Function_Detected:
+        case Function_Alarmed:
+        case Function_PcCorpus:
+        case Function_PcVampire:
+        case Function_Attacked:
+        case Function_TalkedToPc:
+        case Function_ShouldAttack:
+        case Function_Werewolf:
+            mComparisonType = Comparison_Boolean;
+            break;
+
+        // Integer
+        case Function_Journal:
+        case Function_Item:
+        case Function_Dead:
+        case Function_RankLow:
+        case Function_RankHigh:
+        case Function_RankRequirement:
+        case Function_Reputation:
+        case Function_PcReputation:
+        case Function_PcLevel:
```

Comment:
Should I change the valid range for player/actor level to be greater or equal to 1?